File: backend-ruby/app/models/nguoi_dung.rb
class NguoiDung < ApplicationRecord
  self.table_name = 'NguoiDung'
  self.primary_key = 'NguoiDungID'
  belongs_to :vai_tro, class_name: "VaiTro", foreign_key: "VaiTroID"
  has_one :sinh_vien, class_name: "SinhVien", foreign_key: "NguoiDungID"
  has_many :phan_phong_records, class_name: "PhanPhong", foreign_key: "NguoiPhanID"
  has_many :diem_danh_records, class_name: "DiemDanh", foreign_key: "NguoiGhiNhanID"
  has_many :nhat_ky_records, class_name: "NhatKy", foreign_key: "NguoiDungID"
end

File: backend-csharp/Models/NguoiDung.cs
using System.Text.Json.Serialization;

namespace backend_csharp.Models;

public class NguoiDung
{
    public int NguoiDungID { get; set; }
    public string Email { get; set; } = string.Empty;
    public string MatKhauHash { get; set; } = string.Empty;
    public string HoTen { get; set; } = string.Empty;
    public int VaiTroID { get; set; }
    public bool TrangThai { get; set; } = true;
    public DateTime TaoLuc { get; set; }
    public DateTime? CapNhatLuc { get; set; }
    [JsonIgnore] public VaiTro? VaiTro { get; set; }
    [JsonIgnore] public SinhVien? SinhVien { get; set; }
    [JsonIgnore] public ICollection<PhanPhong> PhanPhongRecords { get; set; } = [];
    [JsonIgnore] public ICollection<DiemDanh> DiemDanhRecords { get; set; } = [];
    [JsonIgnore] public ICollection<NhatKy> NhatKyRecords { get; set; } = [];
}
